pkg/detectors/alibaba/alibaba.go

```diff
@@ -17,68 +24,109 @@ type Scanner struct{}
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	keyPat    = regexp.MustCompile(`\b(LTAI[a-zA-Z0-9]{17,21})[\"' ;\s]*`)
-	secretPat = regexp.MustCompile(`\b([a-zA-Z0-9]{30})\b`)
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\b([a-zA-Z0-9]{30})\b`)
+	idPat  = regexp.MustCompile(`\b(LTAI[a-zA-Z0-9]{17,21})[\"';\s]*`)
 )
 
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
 func (s Scanner) Keywords() []string {
-	return []string{"LTAI"}
+	return []string{"alibaba"}
+}
+
+func randString(n int) string {
+	rand.Seed(time.Now().UnixNano())
+	const alphanum = "0123456789abcdefghijklmnopqrstuvwxyz"
+	var bytes = make([]byte, n)
+	rand.Read(bytes)
+	for i, b := range bytes {
+		bytes[i] = alphanum[b%byte(len(alphanum))]
+	}
+	return string(bytes)
+}
+
+func GetSignature(input, key string) string {
+	key_for_sign := []byte(key)
+	h := hmac.New(sha1.New, key_for_sign)
+	h.Write([]byte(input))
+	return base64.StdEncoding.EncodeToString(h.Sum(nil))
+}
+func buildStringToSign(input string) string {
+	filter := strings.Replace(input, "+", "%20", -1)
+	filter = strings.Replace(filter, "*", "%2A", -1)
+	filter = strings.Replace(filter, "%7E", "~", -1)
```

Comment:
Are the parameters swapped here? It looks like the other two are swapping the symbol for the url encoded version.